src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -84,33 +97,51 @@ class SetAppIconRequestTest
     msg_params[am::strings::app_id] = kAppId;
     msg_params[am::strings::cmd_icon] = 1;
     msg_params[am::strings::cmd_icon][am::strings::value] = "10";
-    (*msg)[am::strings::msg_params] = msg_params;
+    msg_params[am::strings::sync_file_name] = kSyncFileName;
+    msg_params[am::strings::params][am::strings::connection_key] =
+        kConnectionKey;
+    (*msg_temp)[am::strings::msg_params] = msg_params;
 
-    return msg;
+    return msg_temp;
   }
-};
 
-TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
-  const std::string file_path = "file_path";
-  MessageSharedPtr msg_vr = CreateFullParamsUISO();
-  (*msg_vr)[am::strings::params][am::strings::connection_key] = kConnectionKey;
-  (*msg_vr)[am::strings::msg_params][am::strings::sync_file_name]
-           [am::strings::value] = file_path;
+  void SetUp() OVERRIDE {
+    const std::string directory = kAppStorageFolder + "/" + kAppFolder;
+    const std::string full_file_path = directory + "/" + kSyncFileName;
+    file_system::CreateDirectoryRecursively(directory);
```

Comment:
@BSolonenko Don't we need to check that directory was created successfully?

Reply:
@AByzhynar  Fixed in https://github.com/smartdevicelink/sdl_core/pull/1974/commits/ce3bce3dc25e2692cf0256f39dffb6b98effa327